app/scripts/controllers/createPersistentVolumeClaim.js

```diff
@@ -55,29 +55,40 @@ angular.module('openshiftConsole')
           }
         };
 
-      function generatePersistentVolumeClaim() {
-        var pvc = {
-          kind: "PersistentVolumeClaim",
-          apiVersion: "v1",
-          metadata: {
-            name: $scope.claim.name,
-            labels: {}
-          },
-          spec: {
-            resources: {
-              requests:{}
+         function generatePersistentVolumeClaim() {
+          var pvc = {
+            kind: "PersistentVolumeClaim",
+            apiVersion: "v1",
+            metadata: {
+              name: $scope.claim.name,
+              labels: {}
+            },
+            spec: {
+              resources: {
+                requests:{}
 
+              },
+              selector:{
+                matchLabels:{}
+              }
             }
-          }
-        };
-
-        pvc.spec.accessModes = [$scope.claim.accessModes || "ReadWriteOnce"] ;
-        var unit =  $scope.claim.unit || "Mi";
-        pvc.spec.resources.requests.storage = $scope.claim.amount + unit;
-
-        return pvc;
-      }
+          };
 
+          pvc.spec.accessModes = [$scope.claim.accessModes || "ReadWriteOnce"] ;
+          var unit =  $scope.claim.unit || "Mi";
+          pvc.spec.resources.requests.storage = $scope.claim.amount + unit;
+          if ($scope.claim.selectedLabels.length > 0)
+          {
+            var arrayLength = $scope.claim.selectedLabels.length;
+            for (var ii = 0; ii < arrayLength; ii++) {
+              var label = $scope.claim.selectedLabels[ii];
+              var keyName = label["name"];
+              var keyValue = label["value"];
+              pvc.spec.selector.matchLabels[keyName] = keyValue;
+            }
```

Comment:
Suggest using lodash here, which is more concise. Should be able to replace the entire `if` block with

```js
_.each($scope.claim.selectedLabels, function(label) {
  pvc.spec.selector.matchLabels[label.name] = label.value;
});
```